tfx/src/main/scala/com/spotify/noether/tfx/TfmaConverter.scala

```diff
@@ -29,35 +30,72 @@ trait TfmaConverter[A, B, T <: Aggregator[A, B, _]] {
 
 object TfmaConverter {
 
-  implicit val binaryConfusionMatrixConverter
-    : TfmaConverter[BinaryPred, Map[(Int, Int), Long], BinaryConfusionMatrix] =
-    new TfmaConverter[BinaryPred, Map[(Int, Int), Long], BinaryConfusionMatrix] {
-      override def convertToTfmaProto(underlying: BinaryConfusionMatrix)
-        : Aggregator[BinaryPred, Map[(Int, Int), Long], MetricsForSlice] =
-        underlying
-          .andThenPresent { matrix =>
-            MetricsForSlice
-              .newBuilder()
-              .setSliceKey(SliceKey.getDefaultInstance)
-              .putMetrics(
-                "Noether_BinaryConfusionMatrix",
-                MetricValue
-                  .newBuilder()
-                  .setConfusionMatrixAtThresholds(
-                    newBuilder()
-                      .setMatrices(0,
-                                   ConfusionMatrixAtThreshold
-                                     .newBuilder()
-                                     .setThreshold(underlying.threshold)
-                                     //                  .setFalseNegatives(matrix)))
-                                     .build())
-                      .build())
-                  .build()
-              )
-              .build()
-          }
+  private def denseMatrixToMetric(matrix: DenseMatrix[Long], threshold: Option[Double] = None)
+  : MetricsForSlice = {
+    val tp = matrix.valueAt(1, 1).toDouble
+    val tn = matrix.valueAt(0,0).toDouble
+    val fp = matrix.valueAt(1, 0).toDouble
+    val fn = matrix.valueAt(0, 1).toDouble
+
+    val cmBuilder = ConfusionMatrixAtThreshold.newBuilder()
+      .setFalseNegatives(fn)
+      .setFalsePositives(fp)
+      .setTrueNegatives(tn)
+      .setTruePositives(tp)
+      .setPrecision(tp / (tp + fp))
+      .setRecall(tp / (tp + fn))
+
+    threshold match {
```

Comment:
Can we replace this with `threshold.forEach(t => cmbuilder.setThreshold(t))`?

Reply:
Nice, can do